cmd/triggerRun/cmd/root.go

```diff
@@ -43,12 +54,57 @@ var (
 func init() {
 	rootCmd.Flags().StringVarP(&triggerFile, "triggerFile", "t", "", "Path to trigger yaml file")
 	rootCmd.Flags().StringVarP(&httpPath, "httpPath", "r", "", "Path to body event")
+	rootCmd.PersistentFlags().StringVarP(&kubeconfig, "kubeconfig", "k", "", "absolute path to the kubeconfig file")
 }
 
 func rootRun(cmd *cobra.Command, args []string) {
 	// TODO: Not implemented
 }
 
+func trigger(w io.Writer, triggerFile, httpPath string) error {
+	// Read HTTP request.
+	r, err := readHTTP(httpPath)
+	if err != nil {
+		return fmt.Errorf("error reading HTTP file: %w", err)
+	}
+	defer r.Body.Close()
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return fmt.Errorf("error reading HTTP body: %w", err)
+	}
+
+	// Read triggers.
+	triggers, err := readTrigger(triggerFile)
+	if err != nil {
+		return fmt.Errorf("error reading triggers: %w", err)
+	}
+
+	client, err := getKubeClient(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("fail to get Kubenetes client: %w", err)
+	}
+
+	logger, _ := zap.NewProduction()
+	s := sink.Sink{
+		Logger: logger.Sugar(),
+	}
+
+	eventID := template.UID()
+	for _, tri := range triggers {
+		eventLog := s.Logger.With(zap.String(triggersv1.EventIDLabelKey, eventID))
```

Comment:
looks like the  sink is only used here to initialize the the eventLog variable below. We can simplify this by getting rid of the sink and just directly using the logger:
`logger.Sugar().With(zap.String(triggersv1.EventIDLabelKey, eventID))`